src/ParallelAlgorithms/Amr/Projectors/Protocols/Mutator.hpp

```diff
@@ -0,0 +1,47 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+
+#include "Utilities/ProtocolHelpers.hpp"
+
+namespace amr::protocols {
+
+/// \brief A DataBox mutator used in AMR actions
+///
+/// A class conforming to this protocol can be used as the template argument for
+/// a call to db::mutate_apply(const gsl::not_null<DataBox<BoxTags>*> box,
```

Comment:
I think it would be clearer to focus these docs on helping the user implement a mutator, rather than being too technical. I think the fact that `db::mutate_apply` is invoked could be nice to mention, but doesn't help the user much because then they have to look up how mutate_apply works and how it fits together with this. I suggest:

- Start with explaining that mutators get called when refining elements.
- The explain the compile-time interface, highlighting that return tags are mutated, argument tags are additional args, and you need three apply functions that take exactly the return tags, arg tags, and one extra arg.
- Add one or more examples.